### ISSUE TITLE
[Haskell] Spacemacs freezes on "Initializing..." haskell-mode

### ISSUE THREAD

**Issue by TxmszLou** [syl20bnr/spacemacs#2896]
On the master branch, opening an ".hs" file or manually start haskell-mode after spacemacs finishes starting up freezes the window on "Initializing" and consumes cpu. I tried a fresh install of spacemacs on develop branch and get the same result.

## System Info
* OS: archlinux
* Emacs: 24.5-1
* Spacemacs: 0.104.0
* Spacemacs brance: develop

The following layers are included in my configuration file.

```elisp
dotspacemacs-configuration-layer
'(
      auto-completion
      git
      org
      eyebrowse
      search-engine
      themes-megapack
      gnus
      (colors :variables colors-enable-nyan-cat-progress-bar t)
      haskell
      python
      markdown
      scala
      javascript
      c-c++
      html
      racket
      (latex :variables latex-enable-auto-fill t)
      xkcd
)
```

**Comment by TxmszLou** [comment]
Same to this issue

https://github.com/kazu-yamamoto/ghc-mod/issues/583

**Comment by kRITZCREEK** [comment]
What does it say in your Messages buffer after you hit C-g?

**Comment by TxmszLou** [comment]
"ghc-mod: unknown command: `map-file'"

**Comment by kRITZCREEK** [comment]
To fix that error I cloned ghc-mod and installed it from master. If you use stack it's advisable to use the same package snapshot for compiling ghc-mod that your project uses.

**Comment by TxmszLou** [comment]
I am using cabal, I built ghc-mod from master with "cabal install" and configured following https://github.com/kazu-yamamoto/ghc-mod#using-the-development-version but same "Initializing" issue still exists as the issue I referenced to suggests.

ghc version: 7.10.1
ghc-mod version: 5.3.0.0

**Comment by kRITZCREEK** [comment]
I'm sorry. I just started getting the same errors. It was all working yesterday though O.o

**Comment by robbyoconnor** [comment]
What revision?

**Comment by stormpat** [comment]
There is some issues with Emacs 24.5.1 and Spacemacs using the dev branch. You could try using master and see if the problems disappear.  Theres another issue at #2869

**Comment by d12frosted** [comment]
I see this problem as well. Develop branch. All installed using stack.

**Comment by robbyoconnor** [comment]
Maybe try and install 24.4 -- I assume people know how to compile software -- unless on windows...that shit is a royal ***PAIN***

**Comment by TxmszLou** [comment]
Same issue on the master branch (0.103.6)

**Comment by robbyoconnor** [comment]
@TxmszLou -- try with 24.4 - you're going to have to compile it yourself

**Comment by TxmszLou** [comment]
@robbyoconnor Sure, I'm downloading the source code... I don't understand what went wrong though...

**Comment by TxmszLou** [comment]
Same on 24.4, I guess it's ghc-mod's bug which is already addressed [in the up coming milestone](https://github.com/kazu-yamamoto/ghc-mod/milestones/v5.4.1.0)

**Comment by robbyoconnor** [comment]
Sometimes upstream bugs happen :frowning:

**Comment by TxmszLou** [comment]
It seems to be fixed on ghc-mod version 5.4.0.0
